src/doc/src/reference/unstable.md

```diff
@@ -508,14 +508,27 @@ CLI paths are relative to the current working directory.
 * Original Pull Request: [#9322](https://github.com/rust-lang/cargo/pull/9322)
 * Tracking Issue: [#9453](https://github.com/rust-lang/cargo/issues/9453)
 
-The `target-applies-to-host` key in a config file can be used set the desired
-behavior for passing target config flags to build scripts.
-
-It requires the `-Ztarget-applies-to-host` command-line option.
-
-The current default for `target-applies-to-host` is `true`, which will be
-changed to `false` in the future, if `-Zhost-config` is used the new `false`
-default will be set for `target-applies-to-host`.
+Historically, Cargo has respected the `linker` and `runner` options (but
+_not_ `rustflags`) of `[target.<host triple>]` configuration sections
+when building and running build scripts, plugins, and other artifacts
+that are _always_ built for the host platform.
+`-Ztarget-applies-to-host` enables the top-level
+`target-applies-to-host` setting in Cargo configuration files which
+allows users to opt into different (and more consistent) behavior for
+these properties.
+
+When `target-applies-to-host` is unset, or set to `true`, in the
+configuration file, the existing Cargo behavior is preserved (though see
+`-Zhost-config`, which changes that default). When it is set to `false`,
+no options from `[target.<host triple>]` are respected for host
+artifacts. Furthermore, when set to `false`, host artifacts do not pick
+up flags from `RUSTFLAGS` or `[build]`, even if `--target` is _not_
+passed to Cargo.
```

Comment:
If you're going to specifically call out the case where `--target` is not passed I think it's worth pointing out here that it basically doesn't work at all. If `target-applies-to-host` is `false` and you type `cargo build` then this PR makes all of `RUSTFLAGS`, `[build]`, and `[target]` get ignored. That's a bug in Cargo that needs to be fixed one day before this is stabilized, of course, but if it's going to be called out here I think it's worthwhile to be called out accurately.

Reply:
I've added a sentence to that effect.

I'm not sure whether it's a bug in Cargo or "working as intended". My understanding at least is that the desire is for `RUSTFLAGS`, `[build]`, and `[target]` to _only_ ever apply to target artifacts, for `cargo build` to act like `cargo build --target <host triple>`, and for `--target <host triple>` to still treat the target as having a separate configuration from the host, which implies this behavior. But I could be missing something about how you'd like this to work.